api/api_full.go

```diff
@@ -528,7 +528,8 @@ type FullNode interface {
 	StateMarketDeals(context.Context, types.TipSetKey) (map[string]*MarketDeal, error) //perm:read
 	// StateMarketStorageDeal returns information about the indicated deal
 	StateMarketStorageDeal(context.Context, abi.DealID, types.TipSetKey) (*MarketDeal, error) //perm:read
-	// StateGetAllocationForPendingDeal returns the allocation for a given deal ID of a pending deal.
+	// StateGetAllocationForPendingDeal returns the allocation for a given deal ID of a pending deal. Returns nil if
+	// pending allocation is not found.
```

Comment:
Q: Can an allocation be not found even when there's a pending allocation in the market actor? This is an assumption I'm making in the impl below.

Reply:
Unless we've made a programmer error then no, there will always be an allocation there.